app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -133,6 +138,15 @@ def get_vm_nics(instance)
         end
       end
 
+      def get_nic_security_group(nic)
+        security_group = nic.properties.try(:network_security_group)
+        return unless security_group
+
+        resource_group = security_group.id[%r{resourceGroups/(.*?)/}, 1]
+        security_group_name = File.basename(security_group.id)
+        @nsg.get(security_group_name, resource_group)
+      end
+
```

Comment:
@djberg96 - instead of making the request to get the security group again, can you look it up in the @data_index hash?